syntax/glsl.vim

```diff
@@ -26,15 +26,15 @@ syn region  glslPreProc         start="^\s*#\s*\(error\|pragma\|extension\|versi
 syn keyword glslBoolean true false
 
 " Integer Numbers
-syn match glslDecimalInt display "\(0\|[1-9]\d*\)"
-syn match glslOctalInt   display "0\o\+"
-syn match glslHexInt     display "0[xX]\x\+"
+syn match glslDecimalInt display "\(0\|[1-9]\d*\)[uUfF]\?"
+syn match glslOctalInt   display "0\o\+[uUfF]\?"
+syn match glslHexInt     display "0[xX]\x\+[uUfF]\?"
```

Comment:
Floating suffixes are not supported with integers (i.e. `float a = 7f` is not valid GLSL) and with hexadecimal number `F` will become simply hexadecimal digit (e.g. `0x123F` is not float number `291.0` but integer `4671`)